mypy/test/testdmypy.py

```diff
@@ -4,52 +4,47 @@
 import re
 import shutil
 import sys
-import time
-import typed_ast
 
 from typing import Dict, List, Optional, Set, Tuple
 
 from mypy import build
 from mypy import defaults
 from mypy.main import process_options
 from mypy.myunit import AssertionFailure
-from mypy.test.config import test_temp_dir, test_data_prefix
-from mypy.test.data import parse_test_cases, DataDrivenTestCase, DataSuite
+from mypy.test.config import test_temp_dir
+from mypy.test.data import DataDrivenTestCase, DataSuite, has_stable_flags, is_incremental
 from mypy.test.helpers import (
     assert_string_arrays_equal, normalize_error_messages,
     retry_on_error, testcase_pyversion, update_testcase_output,
 )
-from mypy.errors import CompileError
 from mypy.options import Options
 
-from mypy import experiments
 from mypy import dmypy
 
 # List of files that contain test case descriptions.
-files = [
-    'check-enum.test',
-    'check-incremental.test',
-    'check-newtype.test',
-]
+if sys.platform != 'win32':
+    dmypy_files = [
+        'check-enum.test',
+        'check-incremental.test',
+        'check-newtype.test',
+    ]
+else:
+    dmypy_files = []
 
 
 class TypeCheckSuite(DataSuite):
+    files = dmypy_files  # type: List[str]
+    base_path = test_temp_dir  # type: str
+    optional_out = True  # type: bool
 
     @classmethod
-    def cases(cls) -> List[DataDrivenTestCase]:
-        if sys.platform == 'win32':
-            return []  # Nothing here works on Windows.
-        c = []  # type: List[DataDrivenTestCase]
-        for f in files:
-            tc = parse_test_cases(os.path.join(test_data_prefix, f),
-                                  None, test_temp_dir, True)
-            c += [case for case in tc
-                  if cls.has_stable_flags(case) and cls.is_incremental(case)]
-        return c
+    def filter(cls, testcase: DataDrivenTestCase) -> bool:
+        return has_stable_flags(testcase) and is_incremental(testcase)
 
     def run_case(self, testcase: DataDrivenTestCase) -> None:
-        assert self.is_incremental(testcase), "Testcase is not incremental"
-        assert self.has_stable_flags(testcase), "Testcase has varying flags"
+        assert has_stable_flags(testcase), "Testcase is not incremental"
```

Comment:
The message should be "Testcase has varying flags" (the messages seem to have been switched on these two lines).